segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/ChunkStorageManager.java

```diff
@@ -0,0 +1,1375 @@
+        this.closed.set(false);
+    }
+
+
+    /**
+     * Attempts to open the given Segment in read-write mode and make it available for use for this instance of the Storage
+     * adapter.
+     * A single active read-write SegmentHandle can exist at any given time for a particular Segment, regardless of owner,
+     * while a read-write SegmentHandle can coexist with any number of read-only SegmentHandles for that Segment (obtained
+     * by calling openRead()).
+     * This can be accomplished in a number of different ways based on the actual implementation of the Storage
+     * interface, but it can be compared to acquiring an exclusive lock on the given segment).
+     *
+     * @param streamSegmentName Name of the StreamSegment to be opened.
+     * @return A CompletableFuture that, when completed, will contain a read-write SegmentHandle that can be used to access
+     * the segment for read and write activities (ex: read, get, write, seal, concat).
+     * If the segment is sealed, then a Read-Only handle is returned.
+     * <p>
+     * If the operation failed, it will be failed with the cause of the failure. Notable exceptions:
+     * <ul>
+     * <li> StreamSegmentNotExistsException: When the given Segment does not exist in Storage.
+     * </ul>
+     */
+    @Override
+    public CompletableFuture<SegmentHandle> openWrite(String streamSegmentName) {
+        checkInitialized();
+        return execute( () -> {
+            long traceId = LoggerHelpers.traceEnter(log, "openWrite", streamSegmentName);
+            Preconditions.checkNotNull(streamSegmentName, "streamSegmentName");
+            try (MetadataTransaction txn = metadataStore.beginTransaction()) {
+                SegmentMetadata segmentMetadata = (SegmentMetadata) txn.get(streamSegmentName);
+                if (null == segmentMetadata || !segmentMetadata.isActive()) {
+                    throw new StreamSegmentNotExistsException(streamSegmentName);
+                }
+                segmentMetadata.checkInvariants();
+                // This segment was created by an older segment store. Need to start a fresh new chunk.
+                if (segmentMetadata.getOwnerEpoch() < this.epoch) {
+                    log.debug("openWrite:Segment needs ownership change. segment: {} ", segmentMetadata.getName());
+                    claimOwnership(txn, segmentMetadata);
+                }
+                // If created by newer instance then abort.
+                if (segmentMetadata.getOwnerEpoch() > this.epoch) {
+                    throw new StorageNotPrimaryException(streamSegmentName);
+                }
+
+                // This instance is the owner, return a handle.
+                val retValue = SegmentStorageHandle.writeHandle(streamSegmentName);
+                LoggerHelpers.traceLeave(log, "openWrite", traceId, retValue);
+                return retValue;
+            } catch (StorageMetadataVersionMismatchException ex) {
```

Comment:
Why do you even bother to catch if you're just going to rethrow? Can you just bubble it up?

Reply:
fixed